File: motor3d 2.0/Render.vb
Namespace Escena.Renders
    Public MustInherit Class Render
        Inherits ObjetoEscena

        Protected mMotor As Motor3D
        Protected mAncho As Integer
        Protected mAlto As Integer

        Public Event Actualizado(ByRef Sender As Render)

        Public ReadOnly Property Motor As Motor3D
            Get
                Return mMotor
            End Get
        End Property

        Public ReadOnly Property Ancho As Integer
            Get
                Return mAncho
            End Get
        End Property

        Public ReadOnly Property Alto As Integer
            Get
                Return mAlto
            End Get
        End Property

        Public Sub New(ByRef Motor As Motor3D)
            MyBase.New()
            mMotor = Motor
            mAncho = 0
            mAlto = 0
        End Sub

        Public Overridable Sub Redimensionar(ByVal Ancho As Integer, ByVal Alto As Integer)
            mAncho = Ancho
            mAlto = Alto
        End Sub

        Public Overridable Sub Iniciar()
            AddHandler mMotor.Actualizado, AddressOf Actualizar
        End Sub

        Public Overridable Sub Finalizar()
            RemoveHandler mMotor.Actualizado, AddressOf Actualizar
        End Sub

        Protected MustOverride Sub Actualizar(ByRef ZBuffer As ZBuffer)
    End Class
End Namespace

File: motor3d 2.0.ConvertedToC#/Render.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;
namespace Motor3D.Escena.Renders
{
	public abstract class Render : ObjetoEscena
	{

		protected Motor3D mMotor;
		protected int mAncho;

		protected int mAlto;
		public event ActualizadoEventHandler Actualizado;
		public delegate void ActualizadoEventHandler(ref Render Sender);

		public Motor3D Motor {
			get { return mMotor; }
		}

		public int Ancho {
			get { return mAncho; }
		}

		public int Alto {
			get { return mAlto; }
		}

		public Render(ref Motor3D Motor) : base()
		{
			mMotor = Motor;
			mAncho = 0;
			mAlto = 0;
		}

		public virtual void Redimensionar(int Ancho, int Alto)
		{
			mAncho = Ancho;
			mAlto = Alto;
		}

		public virtual void Iniciar()
		{
			mMotor.Actualizado += Actualizar;
		}

		public virtual void Finalizar()
		{
			mMotor.Actualizado -= Actualizar;
		}

		protected abstract void Actualizar(ref ZBuffer ZBuffer);
	}
}

